submissions/TaniaBondarenko/frogger_game/app.js

```diff
@@ -0,0 +1,119 @@
+const borders = {
+    rightBorder: 400,
+    leftBorder: 5,
+    upBorder: 410,
+    downBorder: 5
+};
+
+const someGap = 10;
+
+class GameSprite {
+    constructor (x, y,sprite)  {
+        this.sprite = sprite;
+        this.x = x;
+        this.y = y;
+    }
+
+    render () {
+        ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    };
+}
+
+class Enemy extends GameSprite {
+    constructor(x, y, sprite = "images/enemy-bug.png", speed = "75") {
+        super(x, y, sprite);
+        this.speed = speed; 
+    }
+ 
+    update (dt) {
+        this.x += this.speed * dt;
+        const addSpeed = 40;
+        const pointHideEnemy = 500;
+        if (this.x >= pointHideEnemy) {
+            this.x = -someGap;
+            this.speed = Math.floor(addSpeed + Math.random() * 100);
+        }
+    };
+};
+
+let checkCollisions = function () {
+    allEnemies.forEach(function (enemy) {
```

Comment:
Classes should not refer to any global variables, like global variable `player`, which is an instance of `Player` class, or any other similar.

Although referring to global constants and globals provided by the gaming platform like `Resources` is OK.

Hint: pass a global to a class' constructor, save it as a class' own prop and then use the saved reference across class methods.

Reply:
I've got misunderstandings with this function from the very first time. Ant the same is now. I'm stuck.
 All I do leads the same error `checkCollisions` is not defined.
 Please clarify where should I exactly start with this.

Reply:
Make it a method of a relevant class just like update or render

Reply:
> I've got misunderstandings with this function from the very first time. Ant the same is now. I'm stuck.
> All I do leads the same error `checkCollisions` is not defined.
> Please clarify where should I exactly start with this.

@TaniaBondarenko 
This function is defined as global. And you call it in engine.js file.
If you want to change it to 'Player' class method you have to comment line 82 in file engine.js:
![enginejs](https://user-images.githubusercontent.com/2924306/103088937-22ea0c00-45f5-11eb-9efc-13b80a1dd7ac.png)

Because in this line we call the global 'checkCollisions()' that was defined in app.js

Reply:
@OleksiyRudenko, @andrewklmn I am really appreciate of what you both doing for me. But...
When I put `checkCollisions` method in Player or Enemy class, checkCollisions is not defined. And after I commented appropriate line (82) in engine.js  error disappeared, but method is not working (enemies runs through player and nothing happens).
I've been trying the same when I just started, but it didn't worked or like method of class or like Prototype in function. 
Then I leaved it outside of any of my `sprite`s class/functions because it worked somehow. 
I've read some articles but I don't understand how it should be and why is not working. 
If you can give me any materials that could help, I'll be happy.

Reply:
@TaniaBondarenko 
It is not enough just to put `checkCollisison` to Player method and to comment line 82... 
After these two steps we need to call this new Player method `checkCollision` somewhere in your code.
For example in `update` method of Enemy or Player class.
We just need to fire this check somewhere in code now.

Reply:
@andrewklmn, that's it!
Thank you so much!

Reply:
@TaniaBondarenko please explain how did you manage to translate the original code base in `app.js` from prototypical syntax to es6 class syntax.

Reply:
@OleksiyRudenko, i've tried to use recommendations of  the reviewer. 
But nothing could help me to handle with my `checkcollision` as a method which if a part of `Player` or `Enemy`. 
Why can't it be a some global function beyond the Classes?